java-checks/src/main/java/org/sonar/java/checks/helpers/ReassignmentFinder.java

```diff
@@ -94,13 +94,27 @@ private static Tree getClosestReassignment(SyntaxToken startToken, List<Tree> re
       SyntaxToken reassignmentFirstToken = FirstSyntaxTokenFinder.firstSyntaxToken(reassignment);
       int reassignmentLine = reassignmentFirstToken.line();
       int startLine = startToken.line();
-      if (startLine > reassignmentLine || (startLine == reassignmentLine && startToken.column() > reassignmentFirstToken.column())) {
+      if (startLine > reassignmentLine ||
+          (startLine == reassignmentLine &&
+              startToken.column() > reassignmentFirstToken.column() &&
+              !isInAssignedExpression(startToken, reassignment))) {
         result = reassignment;
       }
     }
     return result;
   }
 
+  private static boolean isInAssignedExpression(SyntaxToken syntaxToken, Tree reassignement) {
+    Tree parent = syntaxToken.parent();
+    while (parent != null && !parent.is(Tree.Kind.EXPRESSION_STATEMENT)) {
+      if(parent == reassignement) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)